### DISCUSSION TITLE
Newbie Help

### DISCUSSION THREAD

**Discussion by bbartling** [empicano/aiomqtt#84] [Q&A] [answered]
Hi I am experimenting around with Node Red and running a mosquito broker on the same Ubuntu instance.

In node red just 2 mqtt blocks for an in and an out:

![image](https://user-images.githubusercontent.com/29737117/138705875-1d4aaf91-8915-4631-a27b-e9c546182bfd.png)

The value of `31` is from the `advanced_example` python script in this repo publishing data to the message bus on topic `floors/#` which seems to work great.

But I am stumped trying to publish a message from node red to the Python script running the `advanced_example`.

For example in node red I can use an inject block to publish a string `Testing123` to topic `floors/get/temps` which works and in my Python script running the advanced example I have an `await client.subscribe("floors/#")`

In the advanced example, I can see the message published on the bus that match a topic filter come through on this function:
```
async def log_messages(messages, template):
    async for message in messages:
        # 🤔 Note that we assume that the message paylod is an
        # UTF8-encoded string (hence the `bytes.decode` call).
```

But I cannot seem to create some Boolean statement to flag a particular message. For example is this legit at all? I am trying to flag the Node Red message published `Testing123` but for what ever reason the `if payload_ == "Testing123"` is never True.

```
async def log_messages(messages, template):
    async for message in messages:
        payload_ = template.format(message.payload.decode())
        print("log_messages def hit ",payload_)
        if payload_ == "Testing123":
            print("YES payload_ == Testing123")
        else:
            print("payload_ not == Testing123")
```

Any tips greatly appreciated!!!!!!!!!!!!!!

![image](https://user-images.githubusercontent.com/29737117/138707679-17e6fc5e-c777-459e-b9c5-1745b8c2ad6f.png)

**Comment by bbartling** [comment, marked as answer]
-https://stackoverflow.com/questions/69713386/node-red-to-asyncio-mqtt-debug?noredirect=1#comment123224834_69713386

**Reply by bbartling** [reply]
This is solved I had somewrong with the `if message.payload.decode() == "Testing123"`

I was missing the `.decode()` ==

**Reply by bbartling** [reply]
I have to do some research is MQTT messagse always "strings only" or if json can be used too?

**Reply by frederikaalund** [reply]
Great that you got it to work. :+1: The MQTT message payload is binary. asyncio-mqtt does not make any assumptions about the payload beyond that. :)

In other words, you can send and receive any kind of data. Just make sure to encode the data to binary before you send it, and, decode the binary to your type again after you receive it.
